File: unittest/t_if_add_ts.py
#!/usr/bin/python
import test_utils as ut
import numpy as np
# import nwb

from nwb import nwb_file
from nwb import nwb_utils as utils


# test opening file in append mode
# TESTS creating a module
# TESTS creating an interface
# TESTS adding a timeseries to an interface

def test_file():
    if __file__.startswith("./"):
        fname = "s" + __file__[3:-3] + ".nwb"
    else:
        fname = "s" + __file__[1:-3] + ".nwb"
    create_iface_series(fname, True)
    name1 = "Ones"
    ut.verify_timeseries(fname, name1, "processing/test module/BehavioralEvents", "TimeSeries")


def create_iface_series(fname, newfile):
    settings = {}
    settings["file_name"] = fname
    if newfile:
        # settings["identifier"] = nwb.create_identifier("interface timeseries example")
        settings["identifier"] = utils.create_identifier("interface timeseries example")
        # settings["overwrite"] = True
        settings["mode"] = "w"
        settings["start_time"] = "Sat Jul 04 2015 3:14:16"
        settings["description"] = "Test interface timeseries file"
    else:
        # settings["modify"] = True
        settings["mode"] = "r+"
    # neurodata = nwb.NWB(**settings)
    f = nwb_file.open(**settings)
    #
#     mod = neurodata.create_module("test module")
#     iface = mod.create_interface("BehavioralEvents")
#     ts = neurodata.create_timeseries("TimeSeries", "Ones")
#     ts.set_data(np.ones(10), unit="Event", conversion=1.0, resolution=float('nan'))
#     ts.set_value("num_samples", 10)
#     ts.set_time(np.arange(10))
#     iface.add_timeseries(ts)
#     iface.finalize()
#     mod.finalize()
    
    
    mod = f.make_group("<Module>", "test module")
    iface = mod.make_group("BehavioralEvents")
    ts = iface.make_group("<TimeSeries>", "Ones")
    ts.set_dataset("data", np.ones(10), attrs={'unit':"Event",
        "conversion":1.0, "resolution": float('nan')})
    ts.set_dataset("num_samples", 10)
    ts.set_dataset("timestamps",np.arange(10))
    
    #
    # neurodata.close()
    f.close()

test_file()
print("%s PASSED" % __file__)

File: matlab_bridge/matlab_unittest/t_if_add_ts.m
function t_if_add_ts(verbosity)

% test opening file in append mode
% TESTS creating a module
% TESTS creating an interface
% TESTS adding a timeseries to an interface

script_base_name = mfilename();
script_name = [script_base_name '.m'];
fname = [regexprep(script_base_name, '^t_', 's_') '.nwb'];
if nargin < 1
    % default, display all information. Other options are: 'none', 'summary'
    verbosity = 'all';
end


function test_file()
    create_iface_series(fname, true);
    name1 = 'Ones';
    test_utils.verify_timeseries(fname, name1, 'processing/test module/BehavioralEvents', 'TimeSeries')
end


function create_iface_series(fname, newfile)
    if newfile
         settings = {'file_name', fname, 'mode', 'w', 'verbosity', verbosity, ...
             'start_time', 'Sat Jul 04 2015 3:14:16' ...
             'identifier', nwb_utils.create_identifier('interface timeseries example'), ...
             'description','Test interface timeseries file'};
    else
         settings = {'file_name', fname, 'mode', 'r+'};
    end
    f = nwb_file(settings{:});
  
    %
%     mod = neurodata.create_module('test module')
%     iface = mod.create_interface('BehavioralEvents')
%     ts = neurodata.create_timeseries('TimeSeries', 'Ones')
%     ts.set_data(np.ones(10), unit='Event', conversion=1.0, resolution=float('nan'))
%     ts.set_value('num_samples', 10)
%     ts.set_time(np.arange(10))
%     iface.add_timeseries(ts)
%     iface.finalize()
%     mod.finalize()
    
    
    mod = f.make_group('<Module>', 'test module');
    iface = mod.make_group('BehavioralEvents');
    ts = iface.make_group('<TimeSeries>', 'Ones');
    ts.set_dataset('data', ones([1, 10]), 'attrs', {'unit', 'Event', ...
        'conversion', 1.0, 'resolution', NaN});
    ts.set_dataset('num_samples', 10);
    ts.set_dataset('timestamps',0:9);
    
    %
    % neurodata.close()
    f.close()
end

test_file()
fprintf('%s PASSED\n', script_name);
end

